### PR TITLE
SDI-16 amended date formats, wire up of nomis api call and callback t…

### PR DESCRIPTION
**Pull request by lbennett-moj**
…o VSIP service

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/health/InfoTest.kt [hunk 1]
```diff
@@ -2,10 +2,15 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.integration.health
...
+import org.springframework.test.context.TestPropertySource
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.integration.IntegrationTestBase
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 
+@TestPropertySource(
+  properties =
+  ["spring.autoconfigure.exclude=uk.gov.justice.hmpps.sqs.HmppsSqsConfiguration"]
+)
 class InfoTest : IntegrationTestBase() {
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/health/InfoTest.kt:14]: stop sqs resources being initialised and requiring localstack

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/listeners/PrisonerDomainEventsListener.kt [hunk 2]
```diff
@@ -33,7 +37,8 @@ class PrisonerDomainEventsListener(
 
   data class HMPPSDomainEvent(
     val eventType: String,
-    val additionalInformation: AdditionalInformation,
+    val additionalInformation: AdditionalInformation?,
+    val occurredAt: OffsetDateTime
   )
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/listeners/PrisonerDomainEventsListener.kt:42]: We will look at a better way of dealing with variations on the class

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/PrisonerToNomisTest.kt [hunk 3]
```diff
@@ -37,10 +38,20 @@ class PrisonerToNomisTest : SqsIntegrationTestBase() {
...
+    await untilCallTo { visitsApi.putCountFor("/visits/12/nomis-mapping") } matches { it == 1 }
 
     nomisApi.verify(
       WireMock.postRequestedFor(WireMock.urlEqualTo("/visits"))
         .withRequestBody(WireMock.matchingJsonPath("offenderNo", WireMock.equalTo("A32323Y")))
+        .withRequestBody(WireMock.matchingJsonPath("prisonId", WireMock.equalTo("MDI")))
+        .withRequestBody(WireMock.matchingJsonPath("visitType", WireMock.equalTo("SCON")))
+        .withRequestBody(WireMock.matchingJsonPath("visitRoomId", WireMock.equalTo("Room 1")))
+        .withRequestBody(WireMock.matchingJsonPath("startTime", WireMock.equalTo("2019-12-02T09:00:00")))
+        .withRequestBody(WireMock.matchingJsonPath("issueDate", WireMock.equalTo("2021-03-05")))
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/PrisonerToNomisTest.kt:50]: we could add this to the unit test as well I think

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/health/InfoTest.kt [hunk 4]
```diff
@@ -2,10 +2,15 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.integration.health
...
 import org.junit.jupiter.api.Test
+import org.springframework.test.context.TestPropertySource
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.integration.IntegrationTestBase
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 
+@TestPropertySource(
+  properties =
+  ["spring.autoconfigure.exclude=uk.gov.justice.hmpps.sqs.HmppsSqsConfiguration"]
+)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/health/InfoTest.kt:13]: Maybe we need an annotation of a test proflle for this

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**